src/coreclr/jit/gentree.h

```diff
@@ -4253,12 +4258,19 @@ struct NewCallArg
         return arg;
     }
 
+#ifdef TARGET_WASM
+    static NewCallArg Primitive(GenTree* node, var_types type = TYP_UNDEF, CorInfoType corInfoType = CORINFO_TYPE_UNDEF)
```

Comment:
Instead of this `#ifdef`y way, I think it would better to define a new function:
```
static NewCallArg Primitive(GenTree* node, CorInfoType corInfoType)
{
    assert(corInfoType != CORINFO_TYPE_UNDEF);
    NewCallArg arg = Primitive(node, JITtype2varType(corInfoType));
#ifdef TARGET_WASM
    arg.SignatureCorInfoType = corInfoType;
#endif
    return arg;
}
```
We should use it in `impPopCallArgs` to store the signature type directly. It should allow us to delete the `#ifdef TARGET_WASM` code dealing with signatures for user calls from importer. There is still one case in rationalization (`Rationalizer::RewriteNodeAsCall`) where user calls are created that will need to be handled specially for WASM.

Reply:
I think I'm going to have to allow the var_type to be passed in as well as from `JITtype2varType` or else I hit a problem in `impCheckImplicitArgumentCoercion` with `CORINFO_TYPE_PTR` and `TYP_REF` for `NewCallArg::Primitive(thisForStub, CORINFO_TYPE_PTR)`.  `thisForStub` is of type `TYP_REF`

Reply:
The original code used `CORINFO_TYPE_CLASS` for it, was it changed because that caused other issues?

Reply:
No, I can put that back.  I've lost ~18% of coverage somewhere, so need to see what happened there.

Reply:
It may be due to the removal of signature code. We probably have to keep until the issue with canonical structs is resolved.

Reply:
Yes, that is when it went.  I've put it back